src/WidgetBase.ts

```diff
@@ -421,8 +464,8 @@ export class WidgetBase<P extends WidgetProperties = WidgetProperties, C extends
 		return this._cachedVNode;
 	}
 
-	public invalidate(): void {
-		if (this._renderState === WidgetRenderState.IDLE) {
+	public invalidate(force?: boolean): void {
```

Comment:
In this current PR, this is what's already happening - the forced invalidate merely sets the `_dirty` flag, the actual render won't happen until the next animation frame.